### PR TITLE
chore: cleanup intersphinx links

### PR DESCRIPTION
**Pull request by tswast**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

**Review by plamut — COMMENTED** [review]
Looks good

File: owlbot.py [hunk 1]
```diff
@@ -32,8 +32,6 @@
     intersphinx_dependencies={
         "pandas": "http://pandas.pydata.org/pandas-docs/stable/",
         "geopandas": "https://geopandas.org/",
-        "proto-plus": ("https://proto-plus-python.readthedocs.io/en/latest/", None),
-        "protobuf": ("https://googleapis.dev/python/protobuf/latest/", None),
```
[the remark this answers is not in the document]
Comment (tswast) [reply owlbot.py:36]: It's redundant with the templates from the synthtool repo. I'd rather not add them here + add a line to remove the duplicates (line 59)

File: docs/conf.py [hunk 2]
```diff
@@ -364,10 +364,10 @@
     "google-auth": ("https://googleapis.dev/python/google-auth/latest/", None),
     "google.api_core": ("https://googleapis.dev/python/google-api-core/latest/", None,),
     "grpc": ("https://grpc.github.io/grpc/python/", None),
-    "pandas": ("http://pandas.pydata.org/pandas-docs/stable/", None),
-    "geopandas": ("https://geopandas.org/", None),
     "proto-plus": ("https://proto-plus-python.readthedocs.io/en/latest/", None),
     "protobuf": ("https://googleapis.dev/python/protobuf/latest/", None),
+    "pandas": ("http://pandas.pydata.org/pandas-docs/stable/", None),
```
[the remark this answers is not in the document]
Comment (tswast) [reply docs/conf.py:369]: I remember that too. Maybe we did that in one of the other repos?

[the remark this answers is not in the document]
Comment (tswast) [reply docs/conf.py:369]: Done

**Review by plamut — APPROVED** [review]

**Pull request merged**